docs/en/operations/settings/settings.md

```diff
@@ -1248,7 +1248,9 @@ Possible values:
 Default value: 1.
 
 :::warning
-Disable this setting if you use [max_parallel_replicas](#settings-max_parallel_replicas).
+Disable this setting if you use [max_parallel_replicas](#settings-max_parallel_replicas) without [parallel_replicas_custom_key](#settings-parallel_replicas_custom_key).
+If [parallel_replicas_custom_key](#settings-parallel_replicas_custom_key) is set, disable this setting only if it's used on a cluster with multiple shards containing multiple replicas.
```

Comment:
Maybe check this case and throw an exception (with specified parallel_replicas_custom_key)?

Ideally, it may work in this case as well. We can all the replicas from all the shards, and set replica offset per shard independently (and also not set group_by_no_merge)

Reply:
Well it's not that critical with shards as you said, but it still will disable for one of them parallel replicas because a local plan will just be created.